solidity/contracts/LSP1GraveForwarder.sol

```diff
@@ -48,6 +63,13 @@ contract LSP1GraveForwarder is LSP1UniversalReceiverDelegateUP {
     return tokenAllowlist[msg.sender][token];
   }
 
+  function updateGraveUserCounter(address up) internal {
+    if (!hasJoined[up]) {
+      hasJoined[up] = true;
```

Comment:
this seems redundant. if user has joined they'll already have an entry in `graveVaults` or `tokenAllowlist`. less costly to just have a public function hasJoined that relies on those @Ledespana